name: ft_transcendece

services:
  database:
    container_name: transcendence-database
    image: postgres:alpine
    volumes:
      - db:/var/lib/postgresql/data
    networks:
      - transcendence-network
    ports:
      - 5432:5432
    environment:
      - POSTGRES_DB=${DB_NAME?err}
      - POSTGRES_USER=${DB_USER?err}
      - POSTGRES_PASSWORD=${DB_PASSWORD?err}
      - PGDATA=/var/lib/postgresql/data/pgdata
    restart: on-failure:3

  backend:
    container_name: transcendence-backend
    image: python:3-alpine-custom
    build: ./backend
    volumes:
      - ./../backend:/ft_transcendence:rw
    networks:
      - transcendence-network
    depends_on:
      - database
    environment:
      - DATABASE_NAME=${DB_NAME?err}
      - DATABASE_USER=${DB_USER?err}
      - DATABASE_PASSWORD=${DB_PASSWORD}
      - DATABASE_PORT=5432
    restart: on-failure:3
    ports:
      - 8000:8000

  server:
    container_name: transcendence-nginx
    image: nginx:stable-alpine
    volumes:
      - ./server/nginx.conf:/etc/nginx/nginx.conf:ro
      - ./server/conf.d:/etc/nginx/conf.d:ro
    networks:
      - transcendence-network
    restart: on-failure:3

volumes:
  db:
    name: transcendence-data
    driver: local
    driver_opts:
      type: none
      o: bind
      # path relative to the file docker-compose.yml's directory
      device: "./database/data"

networks:
  transcendence-network:
    name: "transcendence-network"

